src/base/settingsstorage.h

```diff
@@ -62,4 +62,27 @@ public slots:
     mutable QReadWriteLock m_lock;
 };
 
+template<typename T>
+class StorableValue
+{
+public:
```

Comment:
This class has an inconsistency. We set the value using using it as variable, but get the value using it as function:
```c++
someOption = value;
var = someOption();
```

Also you pass default value when get its value. But, IMO, the default value is life-time constant for object of this class.